src/execution/operator/csv_scanner/sniffer/type_detection.cpp

```diff
@@ -373,7 +375,10 @@ void CSVSniffer::SniffTypes(DataChunk &data_chunk, CSVStateMachine &state_machin
 
 				if (row_idx != start_idx_detection &&
 				    (cur_top_candidate == LogicalType::BOOLEAN || cur_top_candidate == LogicalType::DATE ||
-				     cur_top_candidate == LogicalType::TIME || cur_top_candidate == LogicalType::TIMESTAMP)) {
+				     cur_top_candidate == LogicalType::TIME ||
+				     (cur_top_candidate == LogicalType::TIMESTAMP &&
+				      col_type_candidates[col_type_candidates.size() - 2] != LogicalType::TIMESTAMP_TZ) ||
```

Comment:
Can there be only one candidate?

Reply:
`LogicalType::VARCHAR` is always guaranteed to exist

Reply:
I'll add an assertion though :-)